src/gstreamer/gstkvssink.cpp

```diff
@@ -476,6 +492,15 @@ bool kinesis_video_stream_init(GstKvsSink *kvssink, string &err_msg) {
             create_kinesis_video_stream(kvssink);
             break;
         } catch (runtime_error &err) {
+
+            // Don't retry if the error is an internal error
+            if (STRNCMP(INTERNAL_CHECK_PREFIX, err.what(), STRLEN(INTERNAL_CHECK_PREFIX)) == 0) {
```

Comment:
can we move this up?

   ostringstream oss;
    oss << "Failed to create stream. Error: " << err.what();
    err_msg = oss.str();